v2/compiler/argo.go

```diff
@@ -104,6 +105,11 @@ func Compile(jobArg *pipelinespec.PipelineJob, opts *Options) (*wfapi.Workflow,
 	return compiler.wf, nil
 }
 
+func retrieveLastValidString(s string) string {
+	sections := strings.Split(s, "/")
+	return sections[len(sections)-1]
```

Comment:
What happens when sections length is 0? E.g. empty string?

Reply:
I think the section length will be at least one.

Because if I have an empty string "", and I want to split it by `/`, there will be one element in the result `[""]` because there is no splitting happens. 

https://stackoverflow.com/questions/45457397/strings-split-never-returns-nil-or-zero-length-slice

But it is possible that `s` is nil. What do you recommend for returning in this case? Sorry I am relatively new to golang.

Reply:
Thanks! Strings cannot be nil. So you are good on this